src/client/options/mod.rs

```diff
@@ -639,13 +650,38 @@ impl ClientOptions {
     ///   * `wTimeoutMS`: maps to the `w_timeout` field of the `write_concern` field
     ///   * `zlibCompressionLevel`: not yet implemented
     pub async fn parse(s: &str) -> Result<Self> {
-        let parser = ClientOptionsParser::parse(s)?;
+        Self::parse_uri(s, None).await
+    }
+
+    /// Parses a MongoDB connection string into a `ClientOptions` struct.
+    /// If the string is malformed or one of the options has an invalid value, an error will be
+    /// returned.
+    ///
+    /// In the case that "mongodb+srv" is used, SRV and TXT record lookups will be done using the
+    /// provided `ResolverConfig` as part of this method.
+    ///
+    /// The format of a MongoDB connection string is described [here](https://docs.mongodb.com/manual/reference/connection-string/#connection-string-formats).
+    ///
+    /// See the docstring on `ClientOptions::parse` for information on how the various URI options
+    /// map to fields on `ClientOptions`.
+    pub async fn parse_with_resolver_config(
```

Comment:
Good call on this; I hadn't thought of the fact that there would need to be a separate parse method until I started reviewing, so I'm pleased to see you already handled it